src/pytest_workflow/plugin.py

```diff
@@ -178,14 +185,29 @@ def pytest_runtestloop(session: pytest.Session):
     )
 
 
-def pytest_sessionfinish(session: pytest.Session):
-    if not session.config.getoption("keep_workflow_wd"):
-        # The newline is needed otherwise everything looks ugly.
-        print("\nRemoving temporary directories and logs. Use '--kwd' or "
-              "'--keep-workflow-wd' to disable this behaviour.")
+def pytest_sessionfinish(session: pytest.Session, exitstatus: int):
+    def cleanup():
```

Comment:
I'm not such a fan of functions defined in other functions. This could just as well be a helper function defined at the top-level of this file. That also makes testing it easier.

Reply:
You can be too strict in this. In this case it is a helper function that is only used within this function, and only makes sense within this function.
It cannot be tested outside this function.